src/append.jl

```diff
@@ -0,0 +1,168 @@
+"""
+    Arrow.append(file::String, tbl)
+    tbl |> Arrow.append(io_or_file)
+
+Append any [Tables.jl](https://github.com/JuliaData/Tables.jl)-compatible
+`tbl` to an existing arrow formatted file.
+
+Multiple record batches will be written based on the number of
+`Tables.partitions(tbl)` that are provided; by default, this is just
+one for a given table, but some table sources support automatic
+partitioning. Note you can turn multiple table objects into partitions
+by doing `Tables.partitioner([tbl1, tbl2, ...])`, but note that
+each table must have the exact same `Tables.Schema`.
+
+By default, `Arrow.append` will use multiple threads to write multiple
+record batches simultaneously (e.g. if julia is started with `julia -t 8`
+or the `JULIA_NUM_THREADS` environment variable is set).
+
+Supported keyword arguments to `Arrow.append` include:
+  * `compress`: possible values include `:lz4`, `:zstd`, or your own initialized `LZ4FrameCompressor` or `ZstdCompressor` objects; will cause all buffers in each record batch to use the respective compression encoding
+  * `alignment::Int=8`: specify the number of bytes to align buffers to when written in messages; strongly recommended to only use alignment values of 8 or 64 for modern memory cache line optimization
+  * `dictencode::Bool=false`: whether all columns should use dictionary encoding when being written; to dict encode specific columns, wrap the column/array in `Arrow.DictEncode(col)`
+  * `dictencodenested::Bool=false`: whether nested data type columns should also dict encode nested arrays/buffers; other language implementations [may not support this](https://arrow.apache.org/docs/status.html)
+  * `denseunions::Bool=true`: whether Julia `Vector{<:Union}` arrays should be written using the dense union layout; passing `false` will result in the sparse union layout
+  * `largelists::Bool=false`: causes list column types to be written with Int64 offset arrays; mainly for testing purposes; by default, Int64 offsets will be used only if needed
+  * `maxdepth::Int=$DEFAULT_MAX_DEPTH`: deepest allowed nested serialization level; this is provided by default to prevent accidental infinite recursion with mutually recursive data structures
+  * `ntasks::Int`: number of concurrent threaded tasks to allow while writing input partitions out as arrow record batches; default is no limit; to disable multithreaded writing, pass `ntasks=1`
+"""
+function append end
+
+append(file::String; kw...) = x -> append(file::String, x; kw...)
+
+function append(file::String, tbl;
+        largelists::Bool=false,
+        compress::Union{Nothing, Symbol, LZ4FrameCompressor, ZstdCompressor}=nothing,
+        denseunions::Bool=true,
+        dictencode::Bool=false,
+        dictencodenested::Bool=false,
+        alignment::Int=8,
```

Comment:
Same here; we need to match the alignment used in the rest of the file

Reply:
How can the alignment be retrieved from `io_or_file`? Maybe I am missing it, but I do not see that getting stored explicitly anywhere.

Reply:
Ok, now that I think about it and look back at code, I think we can leave this as-is. It's actually valid to use different alignment paddings in different record batches, because the buffers are all specified in full byte counts. So the alignment is "implicit", and shouldn't matter.